evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -675,57 +670,6 @@ def test_automl_supports_time_series_classification(mock_binary_fit, mock_multi_
         assert result['parameters']['pipeline'] == configuration
 
 
-@pytest.mark.parametrize("problem_type", [ProblemTypes.BINARY, ProblemTypes.MULTICLASS])
-@patch('evalml.pipelines.MulticlassClassificationPipeline.fit')
-@patch('evalml.pipelines.MulticlassClassificationPipeline.score')
-@patch('evalml.pipelines.BinaryClassificationPipeline.fit')
-@patch('evalml.pipelines.BinaryClassificationPipeline.score')
-def test_automl_pickle_generated_pipeline(mock_binary_score, mock_binary_fit, mock_multi_score, mock_multi_fit,
-                                          problem_type, X_y_binary, X_y_multi):
-    mock_binary_score.return_value = {"Log Loss Binary": 1.0}
-    mock_multi_score.return_value = {"Log Loss Multiclass": 1.0}
-    if problem_type == ProblemTypes.BINARY:
-        X, y = X_y_binary
-        pipeline = GeneratedPipelineBinary
-
-    elif problem_type == ProblemTypes.MULTICLASS:
-        X, y = X_y_multi
-        pipeline = GeneratedPipelineMulticlass
-
-    a = AutoMLSearch(X_train=X, y_train=y, problem_type=problem_type)
-    a.search()
-
-    for i, row in a.rankings.iterrows():
-        assert a.get_pipeline(row['id']).__class__ == pipeline
-        assert pickle.loads(pickle.dumps(a.get_pipeline(row['id'])))
-
-
-@pytest.mark.parametrize('problem_type', [ProblemTypes.TIME_SERIES_MULTICLASS, ProblemTypes.TIME_SERIES_BINARY])
-@patch('evalml.pipelines.TimeSeriesMulticlassClassificationPipeline.score')
-@patch('evalml.pipelines.TimeSeriesBinaryClassificationPipeline.score')
-@patch('evalml.pipelines.TimeSeriesMulticlassClassificationPipeline.fit')
-@patch('evalml.pipelines.TimeSeriesBinaryClassificationPipeline.fit')
-def test_automl_time_series_classification_pickle_generated_pipeline(mock_binary_fit, mock_multi_fit,
```

Comment:
@jeremyliweishih I agree we should delete these tests but can we add coverage that `get_pipeline` works as expected? 

I think it can be in `test_automl.py` and it can be simple and only test one problem type. I think just checking against the rankings table should suffice:

```python
for _, ranking in automl.rankings.iterrows():
    pl = automl.get_pipeline(ranking.id)
    assert pl.parameters == ranking.parameters
    assert pl.name == ranking.pipeline_name
```